src/Leaflet.Renderer.SVG.Tile.js

```diff
@@ -39,6 +39,16 @@ L.SVG.Tile = L.SVG.extend({
 		}
 	},
 
+	removeFrom: function (map) {
+		if (this.options.interactive) {
+			for (var i in this._layers) {
+				var layer = this._layers[i];
+				delete this._map._targets[L.stamp(layer._path)];
```

Comment:
I haven't looked closely at this, but surely there must be a way to remove the layers from the map's `_targets` without actually fiddling with this internal variable?

VectorGrid is pretty low level, so using Leaflet's internals directly might be ok, but only where strictly necessary.